src/update/UpdateManager.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "VersionId.h"
+#include "VersionPatch.h"
+#include "ServerCommunication.h"
+#include "UpdateDataParser.h"
+
+#include "Project.h"
+#include "Prefs.h"
+
+#include <wx/string.h>
+#include <wx/event.h>
+#include <wx/timer.h>
+
+class UpdateManager final : public wxEvtHandler
+{
+public:
+	UpdateManager(AudacityProject& project);
+	~UpdateManager();
+
+	void getUpdates();
+
+	void enableNotification(bool enable);
+	bool isNotificationEnabled();
+
+	VersionPatch getVersionPatch() const;
+
+	void OnTimer(wxTimerEvent& event);
+
+	DECLARE_EVENT_TABLE()
+
+private:
+	VersionPatch mVersionPatch;
+
+	wxWindow* mParent;
```

Comment:
This plain C pointer is dangerous.  It might become a dangling pointer, because the UpdateManager outlives the window.

wxWeakRef<wxWindow> would be safer, but you would still need to check whether it is null whenever you use it.

But better to eliminate this pointer entirely.